### PR TITLE
fix: migrate firewall to allowedHosts option

### PR DESCRIPTION
**Pull request by anshumanv**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [x] This is a **test update**
- [ ] This is a **docs update**
- [x] This is a **metadata update**

### For Bugs and Features; did you add new tests?
existing 

### Motivation / Use-Case
add more verbose option, firewall was misleading

### Breaking Changes
yep

### CODE REVIEW

File: lib/utils/normalizeOptions.js [hunk 1]
```diff
@@ -115,9 +115,12 @@ function normalizeOptions(compiler, options) {
 
   options.devMiddleware = options.devMiddleware || {};
 
-  if (typeof options.firewall === 'undefined') {
-    // firewall is enabled by default
-    options.firewall = true;
+  if (typeof options.allowedHosts === 'undefined') {
+    // allowedHosts is enabled by default
+    options.allowedHosts = true;
```
[the remark this answers is not in the document]
Comment (anshumanv) [reply lib/utils/normalizeOptions.js:120]: allowed by default means only localhost is allowed by default I think

[the remark this answers is not in the document]
Comment (alexander-akait) [reply lib/utils/normalizeOptions.js:120]: Nope, `allowedHosts: true` means allow all hosts, by default we should have `allowedHosts: false`, only localhost and if you set `host: 'value'` option allowed

[the remark this answers is not in the document]
Comment (anshumanv) [reply lib/utils/normalizeOptions.js:120]: lets fix it

**Comment by anshumanv** [comment]
@alexander-akait done

File: bin/cli-flags.js [hunk 2]
```diff
@@ -405,7 +405,7 @@ module.exports = {
       description: 'The public hostname/ip address of the server.',
     },
     {
-      name: 'firewall',
+      name: 'allowed-hosts',
```
Comment (snitin315) [inline bin/cli-flags.js:408]: Can we add tests for `--allowed-hosts` in `cli.test.js`?

Comment (anshumanv) [reply bin/cli-flags.js:408]: yep lets add it

Comment (anshumanv) [reply bin/cli-flags.js:408]: fixed

File: lib/options.json [hunk 3]
```diff
@@ -331,11 +331,14 @@
       "type": "object",
       "description": "Provide options to webpack-dev-middleware which handles webpack assets. https://webpack.js.org/configuration/dev-server/#devserverdevmiddleware"
     },
-    "firewall": {
+    "allowedHosts": {
```
Comment (snitin315) [inline lib/options.json:334]: Need to update link in description too.

**Comment by anshumanv** [comment]
All done and rebased 👍

File: test/server/Server.test.js [hunk 4]
```diff
@@ -301,12 +301,12 @@ describe('Server', () => {
       });
     });
 
-    it('should always allow any host if options.firewall is disabled', () => {
+    it('should always allow any host if options.allowedHosts is enabled', () => {
```
Comment (alexander-akait) [inline test/server/Server.test.js:304]: Let's move test for `allowedHosts` to own test file, in `Server.test.js` should be only API tests in future

Comment (anshumanv) [reply test/server/Server.test.js:304]: yep lets do it

Comment (anshumanv) [reply test/server/Server.test.js]: done @alexander-akait

**Review by snitin315 — DISMISSED** [review]

**Comment by anshumanv** [comment]
@snitin315 need some more changes, wip 👍

**Comment by alexander-akait** [comment]
Finish your work

**Comment by alexander-akait** [comment]
@snitin315 I think we can finish CLI options, in future when we add new options we need to add them in CLI, do you need help?

**Review by snitin315 — APPROVED** [review]
Looks Good!

**Comment by snitin315** [comment]
> @snitin315 I think we can finish CLI options, in future when we add new options we need to add them in CLI, do you need help?

Yeah, any help finishing tests is appreciated 👍🏻 .

**Pull request merged**